be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -276,7 +282,12 @@ Status SegmentIterator::_init_impl(const StorageReadOptions& opts) {
     _file_reader = _segment->_file_reader;
     _opts = opts;
     _col_predicates.clear();
+
     for (auto& predicate : opts.column_predicates) {
+        if (!_segment->can_apply_predicate_safely(predicate->column_id(), predicate, *_schema,
+                                                  _opts.io_ctx.reader_type)) {
+            continue;
```

Comment:
Now it's safe, since only variant's predicate will `continue` here, and let the expr to handle such filter like inverted index `match` does.It could be a common usage for later `light schema change data type`

Reply:
Add CHECK in `can_apply_predicate_safely ` to make sure `continue` in variant case